clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -730,6 +730,15 @@ void CodeGenFunction::EmitOpenCLKernelMetadata(const FunctionDecl *FD,
         llvm::ConstantAsMetadata::get(Builder.getInt32(1))};
     Fn->setMetadata("stall_enable", llvm::MDNode::get(Context, AttrMDArgs));
   }
+
+  if (const SYCLIntelFPGAMaxConcurrencyAttr *A =
```

Comment:
```suggestion
  if (const auto *A =
```
Because the type is spelled out explicitly in the initializer, this is a good time to use `auto`.